build-monitor-plugin/src/main/java/com/smartcodeltd/jenkinsci/plugins/buildmonitor/viewmodel/features/BadgesDetails.java

```diff
@@ -0,0 +1,91 @@
+package com.smartcodeltd.jenkinsci.plugins.buildmonitor.viewmodel.features;
+
+import com.google.common.collect.ImmutableList;
+import com.smartcodeltd.jenkinsci.plugins.buildmonitor.viewmodel.JobView;
+
+import hudson.model.BuildBadgeAction;
+import jenkins.model.Jenkins;
+
+import static com.google.common.collect.Lists.newArrayList;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+
+import org.apache.commons.jelly.JellyContext;
+import org.apache.commons.jelly.JellyException;
+import org.apache.commons.jelly.XMLOutput;
+import org.codehaus.jackson.annotate.JsonValue;
+import org.kohsuke.stapler.MetaClass;
+import org.kohsuke.stapler.WebApp;
+import org.kohsuke.stapler.jelly.JellyClassLoaderTearOff;
+
+/**
+ * @author Jan Molak
```

Comment:
Feel free to leave your name here, I don't want to take credit for your work :-)